File: app/src/main/java/com/imagegallery/list/service/FlickrImageService.java
package com.imagegallery.list.service;

import com.imagegallery.model.PhotoSearchResult;

import io.reactivex.Single;

public class FlickrImageService implements ImageService {

    public static final String FLICKR_BASE_URL = "https://api.flickr.com";

    private final FlickrRetrofitApiService apiService;

    public FlickrImageService(FlickrRetrofitApiService apiService) {
        this.apiService = apiService;
    }

    @Override
    public Single<PhotoSearchResult> listPhotos() {
        return apiService.listPhotos();
    }

    @Override
    public Single<PhotoSearchResult> searchPhotos(String query) {
        String formattedQuery = query.replaceAll("/[,\\s]/", ",");

        return apiService.searchPhotos(formattedQuery);
    }

}

File: app/src/main/java/com/imagegallery/list/service/FlickrImageService.kt
package com.imagegallery.list.service

import com.imagegallery.model.PhotoSearchResult

import io.reactivex.Single

class FlickrImageService(private val apiService: FlickrRetrofitApiService) : ImageService {

    override fun listPhotos(): Single<PhotoSearchResult> {
        return apiService.listPhotos()
    }

    override fun searchPhotos(query: String): Single<PhotoSearchResult> {
        val formattedQuery = query.replace("/[,\\s]/".toRegex(), ",")

        return apiService.searchPhotos(formattedQuery)
    }

    companion object {
        val FLICKR_BASE_URL = "https://api.flickr.com"
    }
}